### PR TITLE
Clearer error message for useless-else-x type messages

### LINKED ISSUE #5598: ``useless-else-on-loop`` should not be raised on `return`
### Bug description

Here's a typical example of W0120 being raised:

```python
def f():
    while True:
        if condition:
            pass
    else:
        pass
```

And this code **doesn't** raise W0120 (correctly):

```python
def f():
    while True:
        if condition:
            break
    else:
        pass
```

But, this code is pretty much identical in function and it **does** raise W0120 (bug):

```python
def f():
    while True:
        if condition:
            return
    else:
        pass
```

The check for W0120 should check for both `break` and `return` in the while block.

### Configuration

_No response_

### Command used

```shell
pylint mytest.py
```

### Pylint output

```shell
************* Module mytest
mytest.py:5:4: W0120: Else clause on loop without a break statement (useless-else-on-loop)
```

### Expected behavior

Issue not raised

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6-dev0
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

See https://github.com/PyCQA/pylint/pull/5614#issuecomment-1023614434 for rationale.

Closes #5598

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
@tushar-deepsource what do you think ?

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "with statement assignment and exception handler assignment.",
         ),
         "R1705": (
-            'Unnecessary "%s" after "return"',
+            'Unnecessary "%s" after "return" remove it and de-indent all the code inside it',
```
Comment (tusharsadhwani) [inline pylint/checkers/refactoring/refactoring_checker.py:242]: I think here should be a comma before remove, in all the cases

Comment (Pierre-Sassoulas) [reply pylint/checkers/refactoring/refactoring_checker.py]: https://github.com/PyCQA/pylint/pull/5736/commits/a281fadc650072b7737f3b0bb6485011c5f78db4

**Comment by Pierre-Sassoulas** [comment]
Converting to draft because this need a separate message for ``else`` or ``elif``, when it's ``elif`` the fix is to simply remove the ``el`` in ``elif``

**Comment by Pierre-Sassoulas** [comment]
@jacobtylerwalls let me know what you think, this is a follow up to #5614

File: ChangeLog [hunk 2]
```diff
@@ -37,6 +37,10 @@ Release date: TBA
 
   Closes #5281
 
+* Better warning messages for useless else or elif when a function return early.
```
Comment (jacobtylerwalls) [inline ChangeLog:40]:
```suggestion
* Better warning messages for useless else or elif when a function returns early.
```

File: doc/whatsnew/2.13.rst [hunk 3]
```diff
@@ -86,6 +86,10 @@ Other Changes
 
   Closes #5648
 
+* Better warning messages for useless else or elif when a function return early.
```
Comment (jacobtylerwalls) [inline doc/whatsnew/2.13.rst:89]:
```suggestion
* Better warning messages for useless else or elif when a function returns early.
```

File: pylint/extensions/check_elif.py [hunk 4]
```diff
@@ -31,7 +31,8 @@ class ElseifUsedChecker(BaseTokenChecker):
     name = "else_if_used"
     msgs = {
         "R5501": (
-            'Consider using "elif" instead of "else if"',
+            'Consider using "elif" instead of "else", replace the "else" by an "elif",'
+            ' then de-indent the code and switch from "if" to "elif" inside the else',
```
Comment (jacobtylerwalls) [inline pylint/extensions/check_elif.py:35]: I find this suggestion hard to follow. If I try to follow it on the first example in the unittests, I get something like this.

source
```python
    if myint > 5:
        pass
    else:
        if myint <= 5:  # [else-if-used]
            pass
        else:
            myint = 3
            if myint > 2:
                if myint > 3:
                    pass
```

"replace the "else" by an "elif"...
```python
    if myint > 5:
        pass
    elif myint <= 5:
            pass
        else:
            myint = 3
            if myint > 2:
                if myint > 3:
                    pass
```

" then de-indent the code..."

```python
    if myint > 5:
        pass
    elif myint <= 5
            pass
    else:
        myint = 3
        if myint > 2:
            if myint > 3:
                pass
```

" and switch from "if" to "elif" inside the else"
```python
    if myint > 5:
        pass
    elif myint <= 5
            pass  <-- wait, wasn't this to be unindented?
    else: <-- turn this to elif?
        myint = 3
        if myint > 2:  <-- turn this to elif?
            if myint > 3:
                pass
```

I appreciate what we want to do here to help folks out, but I think it might not do any good in some cases. Sorry for going through it so step by step but just wanted to show if we're trying to help folks maybe less is more.

**Review by jacobtylerwalls — COMMENTED** [review]
Great, this is very good for the original `no-else-return` and friends; I just have a slight concern about being misleadingly prescriptive with the `else-if-used` refactoring instructions.

File: pylint/extensions/check_elif.py (hunk 4 above)
Comment (Pierre-Sassoulas) [reply pylint/extensions/check_elif.py]: Ha, I made a slight change before seeing this review, let me try to do it step by step again.

Comment (Pierre-Sassoulas) [reply pylint/extensions/check_elif.py]: > remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else"

```python
   if myint > 5:
        pass
    else:
        if myint <= 5:  # [else-if-used]
            pass
        else:
            myint = 3
            if myint > 2:
                if myint > 3:
                    pass
```     
 remove the ""else"",

```python
   if myint > 5:
        pass
        if myint <= 5:  # [else-if-used]
            pass
        else:
            myint = 3
            if myint > 2:
                if myint > 3:
                    pass
``` 
 then de-indent the code 
```python
   if myint > 5:
        pass
    if myint <= 5:  # [else-if-used]
        pass
    else:
        myint = 3
        if myint > 2:
            if myint > 3:
                pass
```      
switch from ""if"" to ""elif"" inside the else"
```python
   if myint > 5:
        pass
  elif myint <= 5:
        pass
  else:
        myint = 3
        if myint > 2:
            if myint > 3:
                pass
```         
Seems to work, but then again I'm the one that wrote the message so of course I understand it correctly 😄

Comment (jacobtylerwalls) [reply pylint/extensions/check_elif.py]: Yeah, this is better, but it's still hard because "switch from "if" to "elif" inside the else" is referring to an else that's already gone.

What about:
`consider using elif instead of else to remove one indentation level`

Comment (Pierre-Sassoulas) [reply pylint/extensions/check_elif.py]: Yeap, that's nicer let's use this.

Comment (jacobtylerwalls) [reply pylint/extensions/check_elif.py]: Maybe:
Consider consolidating if after else to elif

Comment (Pierre-Sassoulas) [reply pylint/extensions/check_elif.py]: I don't understand "consolidating" so I went with ``'Consider using "elif" instead of "else" then "if" to remove one indentation level'``, which is nice and small imo.

**Review by jacobtylerwalls — APPROVED** [review]

**Review by tusharsadhwani — APPROVED** [review]
